lib/request.js

```diff
@@ -40,6 +42,18 @@ module.exports = (options) => {
             });
         });
         request.on('error', reject);
+        request.on('timeout', () => {
+            request.abort();
+            reject(new Error('socket timeout'));
+        });
+        if (options.dropRequestAfter) {
+            setTimeout(() => {
+                if (!(res && typeof res.body === 'string')) {
```

Comment:
I would suggest instead of saving whole response to variable and then checking on one of the fields, to have an explicit flag like `hasRequestEnded` that is set in `bodyStream.on('end',` callback.